concent_api/core/message_handlers.py

```diff
@@ -547,9 +564,24 @@ def handle_send_force_subtask_results_response(client_message):
     provider_public_key = hex_to_bytes_convert(task_to_compute.provider_public_key)
     requestor_public_key = hex_to_bytes_convert(task_to_compute.requestor_public_key)
 
-    validate_golem_message_signed_with_key(
-        task_to_compute,
+    if subtask_results_accepted is not None:
```

Comment:
I think it colud be moved to the `if ... else` block above - not to add another `if ` cblocks

Reply:
Then I would have to move `TaskToCompute` and keys extracting to `if` block above too, do you want it be this way ?

Reply:
I have better idea. Since `ForceSubtaskResultsResponse` is an instance of `TaskMessage`, you can get `TaskToCompute` (and the keys) directly from client message.